microsetta_private_api/repo/consent_repo.py

```diff
@@ -0,0 +1,167 @@
+    def __init__(self, transaction):
+        super().__init__(transaction)
+
+    doc_read_cols = "distinct on (consent_type) consent_id, consent_type, " \
+                    "locale, date_time, consent_content, reconsent_required" \
+
+    doc_write_cols = "consent_id, consent_type, " \
+                     "locale, date_time, consent_content, " \
+                     "account_id, reconsent_required"
+
+    signature_read_cols = "signature_id, consent_type, " \
+                          "consent_audit.date_time AS sign_date, "\
+                          "reconsent_required"
+
+    signature_write_cols = "signature_id, consent_id, " \
+                           "source_id, date_time, parent_1_name, " \
+                           "parent_2_name, deceased_parent, assent_obtainer"
+
+    def create_doc(self, consent):
+        with self._transaction.dict_cursor() as cur:
+            cur.execute("INSERT INTO ag.consent_documents (" +
+                        ConsentRepo.doc_write_cols + ") "
+                        "VALUES( %s, %s, %s, %s, %s, "
+                        "%s, %s) ",
+                        _consent_document_to_row(consent))
+            return cur.rowcount == 1
+
+    def get_all_consent_documents(self):
+        consent_docs = []
+
+        with self._transaction.dict_cursor() as cur:
+            cur.execute("SELECT " + ConsentRepo.doc_read_cols + " FROM "
+                        "consent_documents ORDER BY consent_type DESC, "
+                        "date_time DESC")
+
+            r = cur.fetchall()
+
+            for index in range(0, len(r)):
+                consent_docs.append(_row_to_consent_document(r[index]))
+
+        return consent_docs
+
+    def get_consent_document(self, consent_id):
+        with self._transaction.dict_cursor() as cur:
+            cur.execute("SELECT " + ConsentRepo.doc_read_cols + " FROM "
+                        "ag.consent_documents WHERE "
+                        "ag.consent_documents.consent_id = %s", (consent_id,))
+            r = cur.fetchone()
+            if r is None:
+                return None
+            else:
+                return _row_to_consent_document(r)
+
+    def is_consent_required(self, source_id, consent_type):
+        with self._transaction.dict_cursor() as cur:
+            cur.execute("SELECT " + ConsentRepo.signature_read_cols + " FROM "
+                        "ag.consent_audit INNER JOIN "
+                        "ag.consent_documents ON "
+                        "consent_audit.consent_id = "
+                        "consent_documents.consent_id "
+                        "WHERE consent_audit.source_id = %s and "
+                        "consent_documents.consent_type "
+                        "LIKE %s ORDER BY sign_date DESC",
+                        (source_id, ('%' + consent_type + '%')))
+
+            r = cur.fetchone()
+            if r is None:
+                return True
+            elif r['reconsent_required'] == 0:
```

Comment:
I think it would be more clear to not use an `elif` here as the value being tested and its interpretation is quite different. 

```suggestion
            if r['reconsent_required'] == 0:
```

Reply:
done